src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -99,3 +99,96 @@ SHARED_API int hostfxr_main(const int argc, const pal::char_t* argv[])
+// from an msbuild other than the one bundled by the CLI.
+//
+// Parameters:
+//    exe_dir
+//      The main directory where SDKs are located in sdk\[version]
+//      sub-folders. Pass the directory of a dotnet executable to
+//      mimic how that executable would search in its own directory.
+//      It is also valid to pass nullptr or empty, in which case
+//      multi-level lookup can still search other locations if 
+//      it has not been disabled by the user's environment.
+//
+//    working_dir
+//      The directory where the search for global.json (which can
+//      control the resolved SDK version) starts and proceeds
+//      upwards. This directory may also be searched for SDKs if
+//      multi-level lookup has not been disabled by the user's
+//      environment. This mimics the current working directory's
+//      interpretation when the dotnet exectuable resolves the SDK
+//      for CLI commands.
+//
+//    buffer
+//      The buffer where the resolved SDK path will be written.
+//
+//    buffer_size
+//      The size of the buffer argument in pal::char_t units.
+//
+// Return value:
+//   <0 - Invalid argument
+//   0  - SDK could not be found.
+//   >0 - The number of characters (including null terminator)
+//        required to store the located SDK.
+//
+//   If resolution succeeds and the positive return value is less than
+//   or equal to buffer_size (i.e. the the buffer is large enough),
+//   then the resolved SDK path is copied to the buffer and null
+//   terminated. Otherwise, no data is written to the buffer.
+//
+// String encoding:
+//   Windows     - UTF-16 (pal::char_t is 2 byte wchar_t)
+//   Unix        - UTF-8  (pal::char_t is 1 byte char)
+//
+SHARED_API int32_t hostfxr_resolve_sdk(
+    const pal::char_t* exe_dir,
+    const pal::char_t* working_dir,
+    pal::char_t buffer[],
+    int32_t buffer_size)
+{
+    trace::setup();
+
+    trace::info(_X("--- Invoked hostfxr [commit hash: %s] hostfxr_resolve_sdk"), _STRINGIFY(REPO_COMMIT_HASH));
+
+    if (buffer_size < 0 || (buffer_size > 0 && buffer == nullptr))
+    {
+        trace::error(_X("hostfxr_resolve_sdk received an invalid argument."));
+        return -1;
+    }
+
+    if (exe_dir == nullptr)
+    {
+        exe_dir = _X("");
+    }
+
+    if (working_dir == nullptr)
+    {
+        working_dir = _X("");
+    }
+
+    pal::string_t cli_sdk;
+    if (!fx_muxer_t::resolve_sdk_dotnet_path(exe_dir, working_dir, &cli_sdk))
+    {
+        // fx_muxer_t::resolve_sdk_dotnet_path handles tracing for this error case.
+        return 0; 
+    }
+
+    if (cli_sdk.size() < buffer_size)
+    {
+        size_t length = cli_sdk.copy(buffer, buffer_size - 1);
+        assert(length == cli_sdk.size());
+        assert(length < buffer_size); 
+        buffer[length] = 0;
+    }
+    else
+    {
+        trace::info(_X("hostfxr_resolve_sdk received a buffer that is too small to hold the located SDK path."));
```

Comment:
This should be verbose not info as it's expected that caller will resize the buffer and call again.